src/Foundatio/Utility/DisposableAction.cs

```diff
@@ -24,10 +24,7 @@ public DisposableAction(Action exitAction)
     /// </summary>
     void IDisposable.Dispose()
     {
-        if (_disposed)
-            return;
-
-        _exitAction();
-        _disposed = true;
+        var exitAction = Interlocked.Exchange(ref _exitAction, null);
```

Comment:
Why add the locking here? I've never seen any dispose methods have locking in them. I'm pretty sure the .NET runtime isn't going to call dispose a bunch of times. I suppose you could call it manually a bunch of times in a row, but it still seems like overhead to add locking here.